crates/swc_ecma_transforms_compat/src/es2020/nullish_coalescing.rs

```diff
@@ -192,6 +192,27 @@ impl VisitMut for NullishCoalescing {
             _ => {}
         }
     }
+
+    fn visit_mut_block_stmt_or_expr(&mut self, n: &mut BlockStmtOrExpr) {
+        let vars = self.vars.take();
+        n.visit_mut_children_with(self);
+
+        if !self.vars.is_empty() {
+            if let BlockStmtOrExpr::Expr(expr) = n {
+                // () => expr
+                // () => { return expr; }
+                let span = expr.span();
+                let mut stmts = vec![Stmt::Return(ReturnStmt {
+                    span: DUMMY_SP,
+                    arg: Some(expr.take()),
+                })];
+                stmts.visit_mut_with(self);
```

Comment:
Do we need to revisit? Can't we just inject the variables?

Reply:
Yes, we should avoid to revisit.